shell/platform/common/cpp/accessibility_bridge.h

```diff
@@ -0,0 +1,240 @@
+    /// that
+    ///                                 can be used for querying the
+    ///                                 accessibility information at the time
+    ///                                 the event is fired.
+    virtual void OnAccessibilityEvent(
+        AXEventGenerator::TargetedEvent targeted_event,
+        AccessibilityBridge* bridge) = 0;
+
+    //------------------------------------------------------------------------------
+    /// @brief      Dispatch accessibility action back to the Flutter framework
+    ///
+    /// @param[in]  target              The semantics node id of the action
+    /// target.
+    /// @param[in]  action              The generated flutter semantics action.
+    /// @param[in]  data                Additional data associated with the
+    /// action.
+    /// @param[in]  data_size           The length of the additional data.
+    virtual void DispatchAccessibilityAction(uint16_t target,
+                                             FlutterSemanticsAction action,
+                                             uint8_t* data,
+                                             size_t data_size) = 0;
+  };
+  //------------------------------------------------------------------------------
+  /// @brief      Creates a new instance of a accessibility bridge.
+  ///
+  /// @param[in]  user_data           A custom pointer to the data of your
+  ///                                 choice. This pointer can be retrieve later
+  ///                                 through GetUserData().
+  AccessibilityBridge(std::unique_ptr<AccessibilityBridgeDelegate> delegate,
+                      void* user_data);
+  ~AccessibilityBridge();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Adds a semantics node update to the pending semantics update.
+  ///             Calling this method alone will NOT update the semantics tree.
+  ///             To flush the pending updates, call the CommitUpdates().
+  ///
+  /// @param[in]  node           A pointer to the semantics node update.
+  void AddFlutterSemanticsNodeUpdate(const FlutterSemanticsNode* node);
+
+  //------------------------------------------------------------------------------
+  /// @brief      Adds a custom semantics action update to the pending semantics
+  ///             update. Calling this method alone will NOT update the
+  ///             semantics tree. To flush the pending updates, call the
+  ///             CommitUpdates().
+  ///
+  /// @param[in]  action           A pointer to the custom semantics action
+  /// update.
+  void AddFlutterSemanticsCustomActionUpdate(
+      const FlutterSemanticsCustomAction* action);
+
+  //------------------------------------------------------------------------------
+  /// @brief      Flushes the pending updates and applies them to this
+  ///             accessibility bridge.
+  void CommitUpdates();
```

Comment:
Docs might want to mention whether this mutates any state that would make it safe/unsafe to call at certain times.

E.g. "Calling this with no pending updates fails/does nothing," or "Callers should prefer calling this with as many/few updates as possible" or "updates of a certain size."

This would partly be to give guidance to platform implementations about how they're expected to handle things, e.g. if the platform only supports batches of certain sizes but the API expects the platform impl to do batching.